lib/ansible/modules/network/aos/aos_asn_pool.py

```diff
@@ -0,0 +1,339 @@
+    session: "{{ session_ok }}"
+    name: "my-asn-pool"
+    state: absent
+
+- name: "Load ASN Pool from File(JSON)"
+  aos_asn_pool:
+    session: "{{ session_ok }}"
+    content: "{{ lookup('file', 'resources/asn_pool_saved.json') }}"
+    state: present
+
+- name: "Delete ASN Pool from File(JSON)"
+  aos_asn_pool:
+    session: "{{ session_ok }}"
+    content: "{{ lookup('file', 'resources/asn_pool_saved.json') }}"
+    state: absent
+
+- name: "Load ASN Pool from File(Yaml)"
+  aos_asn_pool:
+    session: "{{ session_ok }}"
+    content: "{{ lookup('file', 'resources/asn_pool_saved.yaml') }}"
+    state: present
+  register: test
+
+- name: "Delete ASN Pool from File(Yaml)"
+  aos_asn_pool:
+    session: "{{ session_ok }}"
+    content: "{{ lookup('file', 'resources/asn_pool_saved.yaml') }}"
+    state: absent
+'''
+
+RETURNS = '''
+name:
+  description: Name of the ASN Pool
+  returned: always
+  type: str
+  sample: Private-ASN-pool
+
+id:
+  description: AOS unique ID assigned to the ASN Pool
+  returned: always
+  type: str
+  sample: fcc4ac1c-e249-4fe7-b458-2138bfb44c06
+
+value:
+  description: Value of the object as returned by the AOS Server
+  returned: always
+  type: dict
+  sample: {'...'}
+'''
+
+import json
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.aos import get_aos_session, find_collection_item, do_load_resource, check_aos_version, content_to_dict
+
+def check_ranges_are_valid(module, ranges):
+
+    i = 1
+    for range in ranges:
+        if not isinstance(range, list) :
+            module.fail_json(msg="Range (%i) must be a list not %s" % (i, type(range)))
+        elif len(range) != 2:
+            module.fail_json(msg="Range (%i) must be a list of 2 members, not %i" % (i, len(range)))
+        elif not isinstance( range[0], int ):
+            module.fail_json(msg="1st element of range (%i) must be integer instead of %s " % (i,type(range[0])))
+        elif not isinstance( range[1], int ):
+            module.fail_json(msg="2nd element of range (%i) must be integer instead of %s " % (i,type(range[1])))
+        elif range[1] <= range[0]:
+            module.fail_json(msg="2nd element of range (%i) must be bigger than 1st " % (i))
+
+        i += 1
+
+    return True
+
+def get_list_of_range(asn_pool):
+    ranges = []
+
+    for range in asn_pool.value['ranges']:
+        ranges.append([ range['first'], range['last']])
+
+    return ranges
+
+def create_new_asn_pool(asn_pool, name, ranges):
+
+    # Create value
+    datum = dict(display_name=name, ranges=[])
+    for range in ranges:
+        datum['ranges'].append(dict(first=range[0],last=range[1]))
+
+    asn_pool.datum = datum
+
+    ## Write to AOS
+    return asn_pool.write()
+
+
+def asn_pool_absent(module, aos, my_pool):
+
+    margs = module.params
+
+    # If the module do not exist, return directly
+    if my_pool.exists is False:
+        module.exit_json(changed=False, name=margs['name'], id='', value={})
```

Comment:
Could you please add an example (in reply to his comment) of what this error would look like.